File: 1217-RelativeSortArray/1217-RelativeSortArray.cs
// Last updated: 3/26/2026, 1:50:30 AM
public class Solution {
    public int[] RelativeSortArray(int[] arr1, int[] arr2) {
    var result = new List<int>(arr2);
    var temp = new List<int>();
	
    for (int i = 0; i < arr1.Length; i++)
    {
        var index = result.IndexOf(arr1[i]);
        if (index != -1)
             result.Insert(index, arr1[i]);
        else
             temp.Add(arr1[i]);
     }

    for (int i = 0; i < arr2.Length; i++)
    {
        result.Remove(arr2[i]);
    }

    temp.Sort();
    result.AddRange(temp);
    return result.ToArray();
    }
}

File: 1217-RelativeSortArray/1217-RelativeSortArray.java
// Last updated: 3/26/2026, 1:50:33 AM
class Solution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        Arrays.sort(arr1);
        int[] result = new int[arr1.length];
        int[] work = new int[1001];
        for(int i=0;i<arr1.length;i++){
            work[arr1[i]]++;
        }
        int k=0;
        for(int i=0;i<arr2.length;i++){
           int j=work[arr2[i]];
            while(j>0 && k<arr1.length){
                result[k++]=arr2[i];
                j--;
                work[arr2[i]]--;
                
            }
        }
        for(int i=0;i<=1000;i++){
            if(work[i]>0){
                int j=work[i];
                while(j>0 && k<arr1.length){
                    result[k++]=i;
                    j--;
                    work[i]--;
                
                }
            }
        }
        
        return result;
    }
}